prow/clonerefs/run.go

```diff
@@ -36,7 +37,57 @@ import (
 	"k8s.io/test-infra/prow/pod-utils/clone"
 )
 
+// Files to be mounted before cloning, this SSH fingerprint is a well known
+// GitHub fingerprint.
+// ref: https://github.com/kubernetes/test-infra/issues/9638
+func mountDefaultGithubFingerprint(dir string) error {
+	var sshConfigFiles = map[string]string{
+		"/etc/ssh/ssh_config": "github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==",
+		"/github_known_hosts": `Host github.com
+	HostName github.com
+	User git
+	UserKnownHostsFile /github_known_hosts
+	StrictHostKeyChecking yes
+	CheckHostIP no`,
+	}
+
+	// Best effort mounting ssh_config and know_hosts file
+	for fp, content := range sshConfigFiles {
+		// Normalize fp
+		fp = path.Join(dir, fp)
+		// Not to add it repeatedly
```

Comment:
Is this just to save one FS write?

Reply:
not only that, this is modifying a global file, just in case this function was invoked repeatedly these contents should not be appended again and again